README.md

```diff
@@ -197,142 +188,130 @@ class Product < ActiveRecord::Base
+To each ORM object,a `highlight_result` attribute is added. This attribute contains the matched token and snippet for the search result.
 
 ```ruby
-json_answer = Contact.raw_search("jon doe")
-p json_answer
-p json_answer['hits']
-p json_answer['facets']
+hits[0].highlight_result
 ```
 
-Search parameters can be specified either through the index's [settings](https://github.com/algolia/algoliasearch-client-ruby#index-settings-parameters) statically in your model or dynamically at search time specifying [search parameters](https://github.com/algolia/algoliasearch-client-ruby#search) as second argument of the `search` method:
+Use the following method to get the raw JSON response from the API without having to reload the objects from the database:
 
 ```ruby
-class Contact < ActiveRecord::Base
-  include AlgoliaSearch
-
-  algoliasearch do
-    attribute :first_name, :last_name, :email
-
-    # default search parameters stored in the index settings
-    minWordSizefor1Typo 4
-    minWordSizefor2Typos 8
-    hitsPerPage 42
-  end
-end
+json_answer = Episode.raw_search("jesse","summary")
 ```
 
+All [search parameters](https://typesense.org/docs/0.21.0/api/documents.html#search) are specified dynamically at search time by passing it as a hash to the `search` method.
+
 ```ruby
-# dynamical search parameters
-p Contact.raw_search('jon doe', { hitsPerPage: 5, page: 2 })
+Episode.raw_search("jesse","summary",{"sort_by"=>"number:asc"})
 ```
 
 ## Backend Pagination
 
-Even if we **highly recommend to perform all search (and therefore pagination) operations from your frontend using JavaScript**, we support both [will_paginate](https://github.com/mislav/will_paginate) and [kaminari](https://github.com/amatsuda/kaminari) as pagination backend.
+This gem supports both [will_paginate](https://github.com/mislav/will_paginate) and [kaminari](https://github.com/amatsuda/kaminari) as pagination backend.
 
-To use <code>:will_paginate</code>, specify the <code>:pagination_backend</code> as follow:
+To use <code>:will_paginate</code>, specify the <code>:pagination_backend</code> like below:
 
 ```ruby
-AlgoliaSearch.configuration = { application_id: 'YourApplicationID', api_key: 'YourAPIKey', pagination_backend: :will_paginate }
+Typesense.configuration = {
+  nodes: [{
+    host: 'localhost',   # For Typesense Cloud use xxx.a1.typesense.net
+    port: 8108,          # For Typesense Cloud use 443
+    protocol: 'http'         # For Typesense Cloud use https
+  }],
+  api_key: 'xyz',
+  connection_timeout_seconds: 2,
+  pagination_backend: :kaminari
+}
 ```
 
-Then, as soon as you use the `search` method, the returning results will be a paginated set:
+When you use the `search` method, the returning results will be a paginated set:
 
 ```ruby
 # in your controller
-@results = MyModel.search('foo', hitsPerPage: 10)
+@results = Episode.search('jesse', 'summary', { per_page: 5, page: params[:page] })
```

Comment:
It looks like the ransack gem(https://github.com/activerecord-hackery/ransack#ransack-search-method) is causing trouble here because of the same search method.